packages/rulesets/src/spectral/functions/put-request-response-scheme.ts

```diff
@@ -0,0 +1,48 @@
+// Verifies if a PUT operation request and response schemas match.
+
+export const putRequestResponseScheme = (putOp: any, _opts: any, ctx: any) => {
+  if (putOp === null || typeof putOp !== "object") {
+    return [];
+  }
+  const path = ctx.path || [];
+  const errors: any = [];
+  if (!putOp.parameters) {
+    return [];
+  }
+  let reqBodySchemaRef = "";
+  const swagger = ctx.document.parserResult.data;
```

Comment:
same with above

Reply:
Got it, could you add a positive test case for it

Reply:
I think you can use the resolved swagger, you can just compare the object references between the request and response schema, as if the parameter is an external reference, your logic will not work